src/data/transforms.py

```diff
@@ -1,169 +1,233 @@
+    'Compose',
+    'ToTensor',
+    'Normalize',
+    'MinMaxScale',
+    'Clip',
+    'Resample',
+    'RandomCrop',
+    'RandomElasticDeform',
+    'RandomHorizontalFlip',
+    'RandomVerticalFlip',
 ]
 
 
-def compose(transforms=None):
+class Compose:
     """Compose several transforms together.
     Args:
-        transforms (Box): The preprocessing and augmentation techniques applied to the data.
-
-    Returns:
-        transforms (Compose): The list of BaseTransform.
+         transforms (sequence of BaseTransform, optional): The preprocessing and augmentation techniques
+            applied to the data (default: None, do not apply any transform).
     """
-    if transforms is None:
-        return None
-
-    _transforms = []
-    for transform in transforms:
-        cls = getattr(src.data.transforms, transform.name)
-        kwargs = transform.get('kwargs')
-        _transforms.append(cls(**kwargs) if kwargs else cls())
-    transforms = Compose(_transforms)
-    return transforms
-
 
-class BaseTransform:
-    """The base class for all transforms.
-    """
+    def __init__(self, transforms=None):
+        if transforms is None:
+            self.transforms = tuple()
+        else:
+            if not all(isinstance(transform, BaseTransform) for transform in transforms):
+                raise TypeError('All of the transforms should be BaseTransform.')
+            self.transforms = tuple(transforms)
 
-    def __init__(self):
-        pass
+    def __call__(self, *imgs, **transforms_kwargs):
+        """
+        Args:
+            imgs (tuple of numpy.ndarray): The images to be transformed.
+            transforms_kwargs (dict): The runtime kwargs for each transforms.
 
-    def __call__(self, *imgs, **kwargs):
-        raise NotImplementedError
+        Returns:
+            imgs (tuple): The transformed images.
+        """
+        for transform in self.transforms:
+            kwargs = transforms_kwargs.get(transform.__class__.__name__, {})
+            transformed = kwargs.pop('transformed', tuple(True for _ in range(len(imgs))))
+            if len(transformed) != len(imgs):
+                raise ValueError('The number of the transformed should be the same as the images.')
+            if not all(_transformed in [True, False] for _transformed in transformed):
+                raise ValueError('All of the transformed should be either True or False.')
+            if not any(transformed):
+                raise ValueError('The transformed should not be all False.')
+
+            if all(transformed):
+                imgs = transform(*imgs, **kwargs)
+            else:
+                transformed_imgs, untransformed_imgs = self._split_imgs(imgs, transformed)
+                transformed_imgs = transform(*transformed_imgs, **kwargs)
+                imgs = self._reassemble_imgs(transformed_imgs, untransformed_imgs, transformed)
+        return imgs
 
     def __repr__(self):
```

Comment:
where is the `print function` called

Reply:
Sorry, I don't understand what's your concerns XD.

Reply:
I mean I don't see the `print(compose)` in any code.

Reply:
I think that depends on the users, we just provide this functionality for printing some useful information ([ref](https://stackoverflow.com/questions/1436703/difference-between-str-and-repr)).